core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -96,7 +97,12 @@ object RequestChannel extends Logging {
       else
         null
 
-    buffer = null
+    //most request types are parsed entirely into objects at this point. for those we can release the underlying buffer.
+    //some (like produce, or any time the schema contains fields of types BYTES or NULLABLE_BYTES) retain a reference
+    //to the buffer. for those requests we cannot release the buffer early, but only when request processing is done.
+    if (requestId != ApiKeys.PRODUCE.id && requestId != ApiKeys.JOIN_GROUP.id && requestId != ApiKeys.SYNC_GROUP.id) {
```

Comment:
Is there any concern that we might lose track of updating this properly? It's not just an issue with adding new request types; it's also a problem if a request type that didn't have `BYTES` or `NULLABLE_BYTES` fields is updated to a version of the schema that does have them.

I'm skeptical that folks will even know about, let alone remember to update, this list if they introduce such a field. Would some sort of static determination based on the full list of schemas be more reliable but equally fast?